### ISSUE TITLE
have a datasets path bug

### ISSUE THREAD

**Issue by xiaohangguo** [ultralytics/ultralytics#2221]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

my data.yaml path:
```
path: ./COCOFireLandStorm  # dataset root dir
train: images/train  # train images (relative to 'path') 128 images
val: images/valid  # val images (relative to 'path') 128 images
test:  # test images (optional)
```

and run:
```
from ultralytics import YOLO

# Load a model
# model = YOLO("yolov8n.yaml")  # build a new model from scratch
model = YOLO("yolov8s.pt")  # load a pretrained model (recommended for training)

# Use the model
results = model.train(data="/media/hang/软件/yolov8_tracking/yolov8/datasets/COCOFireLandStorm/data.yaml", epochs=60)  # train the model
```

get error:
`Dataset '/media/hang/软 件/yolov8_tracking/yolov8/datasets/COCOFireLandStorm/data.yaml' not found ⚠️, missing paths ['/home/hang/datasets/COCOFireLandStorm/images/valid']`

### Additional

Why is my path automatically set to '/home'?

**Comment by maheeetaaa** [comment]
I have the same issue 
Note dataset download directory is '/content/datasets'. You can update this in '/root/.config/Ultralytics/settings.yaml'

I ham using the hub so I dont know how to update this. Also I am training a costum dataset , that has 8 classes, does the model automatically update that too? Because I used to change that from yolov5s.yaml when I used yolo5.

**Comment by maheeetaaa** [comment]
hmmm, I did that but i still had some issues... are you using your local CUDA or training on colab?

**Comment by xiaohangguo** [comment]
yes，i am running yolov8(ultralytics) at the local computer

**Comment by glenn-jocher** [comment]
@xiaohangguo if you are facing issues with the dataset path, you can try modifying the path in the data.yaml file. You can verify that the path is correct as per your local setup. It's expected that the model automatically updates itself according to the number of classes in the dataset. You may also check settings.yaml to confirm the default dataset directory in use. If you're still facing issues after trying these steps, you can provide more details on the specific errors you're encountering, and we can explore further solutions.

**Comment by xiaohangguo** [comment]
> I have the same issue Note dataset download directory is '/content/datasets'. You can update this in '/root/.config/Ultralytics/settings.yaml'
> 
> I ham using the hub so I dont know how to update this. Also I am training a costum dataset , that has 8 classes, does the model automatically update that too? Because I used to change that from yolov5s.yaml when I used yolo5.

I have also used YOLOv5, but now it seems that the data. yaml format for both it and YOLOv8 has changed. Previously
```
train:
valid:
nc:
names:["xxx","xxxx","xxxx"]

```
Now it's like this:
```
path:
train:
valid:
nc:85
names:
    0:'xxx'
    1:'xxx'
```

**Comment by glenn-jocher** [comment]
@xiaohangguo Yes, the data.yaml format has changed for both YOLOv5 and YOLOv8. In the new format, the `nc` parameter specifies the number of classes in the dataset, and the `names` parameter lists the individual class names. In YOLOv8 the `path` parameter specifies the root directory of the dataset. Regarding the default dataset directory, you can check `settings.yaml` to confirm the current directory in use. If you're still encountering issues after trying the above solutions, be sure to provide specific details on the errors you are encountering so we can further assist you.

**Comment by Laughing-q** [comment]
@xiaohangguo using a absolute path here should solve your issue:
```yaml
path: /xxx/COCOFireLandStorm  # dataset root dir
train: images/train  # train images (relative to 'path') 128 images
val: images/valid  # val images (relative to 'path') 128 images
test:  # test images (optional)
```
if you are using a relative path, it'll assume the path is relative to the `datasets_dir` in your settings. use the following commend to check your yolo settings. :) 
```bash
yolo settings
```

**Comment by xiaohangguo** [comment]
> @xiaohangguo using a absolute path here should solve your issue:
> 
> ```yaml
> path: /xxx/COCOFireLandStorm  # dataset root dir
> train: images/train  # train images (relative to 'path') 128 images
> val: images/valid  # val images (relative to 'path') 128 images
> test:  # test images (optional)
> ```
> 
> if you are using a relative path, it'll assume the path is relative to the `datasets_dir` in your settings. use the following commend to check your yolo settings. :)
> 
> ```shell
> yolo settings
> ```

thanks, get it，朝伟

**Comment by xx-xxx** [comment]
How to use my own dataset,  need an images directory, what file contents do i need to put in the valid and train directory.
What's in it the train.txt and label.txt files and how i organiz them?

**Comment by pderrenger** [comment]
@xx-xxx to use your own dataset with YOLOv8, you need an "images" directory containing your training and validation images. 

In the "train" and "valid" directories, you just need to put your images. For example, in the "train" directory, you would put the images that you want to use for training the model. Similarly, in the "valid" directory, you would put the images for validation.

Regarding the "train.txt" and "label.txt" files, you do not need to create these files. YOLOv8 will automatically generate the necessary labels during training based on the directory structure of your dataset. 

You just need to organize your dataset directory as follows:
```
dataset_root  # dataset root directory
├── images
│   ├── train  # training images
│   │   ├── img1.jpg
│   │   ├── img2.jpg
│   │   └── ...
│   └── valid  # validation images
│       ├── img101.jpg
│       ├── img102.jpg
│       └── ...
└── data.yaml  # dataset configuration
```

In the `data.yaml`, you specify the paths to your training and validation images as shown in your previous post. 

After organizing your dataset like this, you can then proceed to train your YOLOv8 model using this dataset.
